src/test/java/net/minecraftforge/debug/misc/RegistryCodecTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.misc;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Mod("registry_codec_test")
+public class RegistryCodecTest
+{
+
+    private static final Logger LOGGER = LogManager.getLogger("Codec Registry Test");
+
+    /**
+     *     This Codec can serialize and deserialize a Pair<Item, Block>
+     *     The result Json (or nbt equivalent) will have this structure :
+     *     {
+     *         "block": "block_registry_name",
+     *         "item": "item_registry_name"
+     *     }
+     */
+    private static final Codec<Pair<Block, Item>> CODEC = RecordCodecBuilder.create(codecInstance -> codecInstance.group(
+            ForgeRegistries.BLOCKS.getCodec().fieldOf("block").forGetter(Pair::getFirst),
+            ForgeRegistries.ITEMS.getCodec().fieldOf("item").forGetter(Pair::getSecond)
+    ).apply(codecInstance, Pair::of));
+
+    public RegistryCodecTest()
+    {
+        FMLJavaModLoadingContext.get().getModEventBus().addListener(this::commonSetup);
+    }
+
+    public void commonSetup(final FMLCommonSetupEvent event)
+    {
```

Comment:
Could you please add a test case for decoding using the compressed JSON ops? Here's one I made to test it myself:
```java
JsonArray jsonCompressed = new JsonArray();
jsonCompressed.add(((ForgeRegistry<Block>) ForgeRegistries.BLOCKS).getID(Blocks.DIAMOND_BLOCK));
jsonCompressed.add(((ForgeRegistry<Item>) ForgeRegistries.ITEMS).getID(Items.DIAMOND_PICKAXE));

DataResult<Pair<Pair<Block, Item>, JsonElement>> resultCompressed = CODEC.decode(JsonOps.COMPRESSED, jsonCompressed);
resultCompressed.resultOrPartial(LOGGER::warn).ifPresent(pair -> LOGGER.info("Decoded {} and {}", pair.getFirst().getFirst().getRegistryName(), pair.getFirst().getSecond().getRegistryName()));
```
You could also pipe in as the input the output of the compressed JSON ops in the last test case instead.

Reply:
Added your test case